translator/src/main/java/org/finos/calmtranslator/translators/K8sModelTranslator.java

```diff
@@ -1,21 +1,89 @@
 package org.finos.calmtranslator.translators;
 
-import io.fabric8.kubernetes.api.model.Pod;
-import io.fabric8.kubernetes.api.model.PodBuilder;
+import io.fabric8.kubernetes.api.model.Namespace;
+import io.fabric8.kubernetes.api.model.NamespaceBuilder;
+import io.fabric8.kubernetes.api.model.Service;
+import io.fabric8.kubernetes.api.model.ServiceBuilder;
+import io.fabric8.kubernetes.api.model.ServicePort;
 import io.fabric8.kubernetes.client.utils.Serialization;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
 import org.finos.calmtranslator.calm.Core;
+import org.finos.calmtranslator.calm.Node;
+import org.finos.calmtranslator.calm.RelationshipType;
+
 public class K8sModelTranslator implements ModelTranslator<String> {
-	@Override
-	public String translate(final Core calmModel) {
-
-		Pod p = new PodBuilder().withNewMetadata()
-				.withName("excj496")
-				.endMetadata()
-				.build();
-		// Get YAML string
-		System.out.println(Serialization.asYaml(p));
-		// Get JSON string
-		System.out.println(Serialization.asJson(p));
-		return null;
-	}
+    @Override
+    public String translate(final Core calmModel) {
+
+        // List of namespaces in calm model
+        List<String> namespaces = new ArrayList<>();
+
+        // Map of Service (unique ID) to Namespace
+        Map <String,String> serviceToNamespaceMap = new HashMap<>();
+
+        //Populate Collections
+        calmModel.getRelationships().forEach(relationship -> {
+            final RelationshipType relationshipType = relationship.getRelationshipType();
+            if (Objects.nonNull(relationshipType.getComposedOf())) {
+                String namespace = relationshipType.getComposedOf().getContainer();
+                namespaces.add(namespace);
+                relationshipType.getComposedOf().getNodes().forEach( node -> {
+                    serviceToNamespaceMap.put(node ,namespace);
+                });
+            }
+        });
+
+        StringBuilder response = new StringBuilder();
+
+        //Build Namespaces
+        response.append(
+            namespaces.stream().map(ns -> createNamespace(ns)).collect(Collectors.joining("")));
+
+        //Build Services 
+        response.append(
+        calmModel.getNodes().stream()
+            .filter(node -> node.getNodeType() == Node.NodeTypeDefinition.SERVICE)
+            .map(service -> createService(
+                    service,serviceToNamespaceMap)).collect(Collectors.joining("")));
+        return response.toString();
```

Comment:
I don't think there is a separation here?
The code looks like it just appends the yaml together, there is no line separation between the components.

Reply:
I just used a simple Collector implementation to join the output. The Fabric8 lib provides the separator as part of its interface. Sample below. We could add another one but that wouldn't be help imho.

```
---
apiVersion: "v1"
kind: "Namespace"
metadata:
  name: "traderx-system"
---
apiVersion: "v1"
kind: "Service"
metadata:
  name: "web-gui"
  namespace: "traderx-system"
spec:
  ports:
  - port: 8080
  selector:
    app: "web-gui"
```

Reply:
Ok this is getting removed in coming changes